recipes/simdjson/0.5.0/conanfile.py

```diff
@@ -0,0 +1,97 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class SimdjsonConan(ConanFile):
+    name = "simdjson"
+    description = "Parsing gigabytes of JSON per second"
+    topics = ("conan", "json", "parser", "simd", "format")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/lemire/simdjson"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "threads": [True, False]}
+    default_options = {'shared': False,
+                       'fPIC': True,
+                       'threads': True}
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+          return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+          return "build_subfolder"
+
+    @property
+    def _supported_cppstd(self):
+        return ["11", "gnu11", "14", "gnu14", "17", "gnu17", "20", "gnu20"]
+
+    def _is_supported_compiler(self):
+        # Try to get by conan. We support more compilers than that.
+        supported_compilers = [("apple-clang", 10), ("gcc", 7.4), ("clang", 6), ("Visual Studio", 15.7)]
+        compiler, version = self.settings.compiler, Version(self.settings.compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd and \
+           not self.settings.compiler.cppstd in self._supported_cppstd:
```

Comment:
I think you can use [tools.check_min_cppstd](https://docs.conan.io/en/latest/reference/tools.html#tools-check-min-cppstd) here.

Reply:
Unfortunately, we cannot. `check_min_cppstd` works only if `cppstd` is set which is not a case for CCI when cppstd is not set. So it is required to explicitly check compiler versions to to avoid CCI broken builds.  Please check a [recent discussion](https://cpplang.slack.com/archives/C41CWV9HA/p1593529376417300?thread_ts=1593523616.395800&cid=C41CWV9HA) for details and proposed improvements.

Reply:
cppstd is not declared by default and most of users don't use it. You can use something similar to https://github.com/conan-io/conan-center-index/blob/a6611a1a6f57353efa2249194eb9e775e4d6141b/recipes/dataframe/all/conanfile.py#L55

Reply:
Ah, sorry, I thought you meant both checks should be replaced with one `check_min_cppstd` call. Replaced the first check with the `check_min_cppstd ` call.